### ISSUE TITLE
pointer freed but not allocated in fx_mix_lab (2.9.9., osx brew build)

### ISSUE THREAD

**Issue by gpciceri** [dtschump/gmic#332]
Hi all,
I'm using 2.9.9 build on OSX (brew package). Please let me know what you need in order to help me further investigate this issue (on windows, I never had a problem).

The command:

/usr/local/bin/gmic -v 3  -i DN-SSC-C9079123.MOV-0.jpeg -fx_tk_photoillustration 0,0.25,0,0.3,0.5,0.5,1,0,5,0,1,0,0,1,0,0,1.2,98.5,5,0.5,0,0,0,0,0,1,0 -o PI-DN-SSC-C9079123.MOV-0.jpeg

the output (last lines)
...
[gmic]-1./fx_tk_photoillustration/*repeat/*local/*if/fx_smooth_anisotropic/*repeat/*local/ac/ Apply command 'repeat 1 smooth 50,0.59999999999999998,0.3,0.5,0.5,0.80,30.00,2.0,0,1 done' on channels 'all' of image [0].
[gmic]-1./fx_tk_photoillustration/*repeat/*local/*if/fx_smooth_anisotropic/ Cut image [0] in range [0,255].
[gmic]-2./fx_tk_photoillustration/*repeat/*local/*if/ Reach 'else' block.
[gmic]-2./fx_tk_photoillustration/*repeat/*local/*if/ End 'if...endif' block.
[gmic]-1./fx_tk_photoillustration/*repeat/*local/fx_mix_lab/*repeat/*local/ Split color and opacity parts of image [0].
[gmic]-1./fx_tk_photoillustration/*repeat/*local/fx_mix_lab/*repeat/*local/ Force image [0] to be in RGB mode.

[gmic] G'MIC encountered a fatal error. Please submit a bug report, at: https://github.com/dtschump/gmic/issues

[gmic] G'MIC encountered a fatal error. Please submit a bug report, at: https://github.com/dtschump/gmic/issues

gmic(1102,0x70000a736000) malloc: *** error for object 0x7fd80b4b0d30: pointer being freed was not allocated

(the image is attached).
![DN-SSC-C9079123 MOV-0](https://user-images.githubusercontent.com/120627/132459851-aee7c7ef-9a23-4ea5-b43f-7f3bc44a6adc.jpeg)

**Comment by dtschump** [comment]
I don't own a Mac, however I did the following tests (on my PC running Ubuntu 20.4) :

- Compile `gmic` with the `fsanitize=address` flag activated (`make debug`), to track possible memory leaks or bad access. Then I've tried the exact command line you put in your post, with the input image you also gave. Unfortunately, everything went right. No errors.
- Recompile `gmic` again for trying it with valgrind (`make valgrind`), and run `valgrind`, again on the same command line. No errors were found:
~~~
==1826714== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
~~~

At this point, this seems to be an issue happening only on Mac. It will be a bit hard for me to debug it, as I don't own a Mac. If anyone can generate a valgrind report that exhibits the bug, that would be really helpful.
